DetectorDescription/DDCMS/interface/DDFilteredView.h

```diff
@@ -87,13 +89,43 @@ namespace cms {
     //! pop current node
     void unCheckNode();
 
+    // Shape of current node
+    bool isABox() const;
+    bool isAConeSeg() const;
+    bool isAPseudoTrap() const;
+    bool isATrapezoid() const;
+    bool isATruncTube() const;
+    bool isATubeSeg() const;
+
+    // Get shape pointer of current node.
+    // Caller must check that current node matches desired type
+    // before calling this function.
+
+    template <class T>
```

Comment:
@cvuosalo - why not return a ```DDShape``` and move all the classes in ```DDFilteredView```?  Then the ```MagGeoBuilderFromDDD::volumeHandle``` class can have a ```template <class T> build(const T&);``` method with specialization for each of the DDShapes, for example: 
```template <> MagGeoBuilderFromDDD::volumeHandle::build<>(const DDFilteredView::DDBox& box);``` etc.

Reply:
This change can be considered after the migration when the old DD is dropped. In migrating the magnetic field geometry builder, my guiding principles were to minimize as much as possible any changes to the old code and to re-use as much as the old code as I could.

Reply:
@ianna I understand you are ok with the proposal of Carl